analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ExistsInsteadOfAny.cs

```diff
@@ -0,0 +1,212 @@
+        goodList.GetList().GetList().GetList()?.GetList()?.Any(x => x > 0);   // Noncompliant
+        goodList.GetList().GetList()?.GetList().GetList().Any(x => x > 0);    // Noncompliant
+        goodList.GetList().GetList()?.GetList().GetList()?.Any(x => x > 0);   // Noncompliant
+        goodList.GetList().GetList()?.GetList()?.GetList().Any(x => x > 0);   // Noncompliant
+        goodList.GetList().GetList()?.GetList()?.GetList()?.Any(x => x > 0);  // Noncompliant
+        goodList.GetList()?.GetList().GetList().GetList().Any(x => x > 0);    // Noncompliant
+        goodList.GetList()?.GetList().GetList().GetList()?.Any(x => x > 0);   // Noncompliant
+        goodList.GetList()?.GetList().GetList()?.GetList().Any(x => x > 0);   // Noncompliant
+        goodList.GetList()?.GetList().GetList()?.GetList()?.Any(x => x > 0);  // Noncompliant
+        goodList.GetList()?.GetList()?.GetList().GetList().Any(x => x > 0);   // Noncompliant
+        goodList.GetList()?.GetList()?.GetList().GetList()?.Any(x => x > 0);  // Noncompliant
+        goodList.GetList()?.GetList()?.GetList()?.GetList().Any(x => x > 0);  // Noncompliant
+        goodList.GetList()?.GetList()?.GetList()?.GetList()?.Any(x => x > 0); // Noncompliant
+    }
+
+    void CheckDelegate(
+        List<int> intList,
+        List<string> stringList,
+        List<ClassA> refList,
+        int[] intArray,
+        string someString,
+        int someInt,
+        int anotherInt,
+        ClassA someRef)
+    {
+        intList.Any(x => x == 0); // Compliant (should raise S6617)
+        intList.Any(x => 0 == x); // Compliant (should raise S6617)
+        intList.Any(x => x == someInt); // Compliant (should raise S6617)
+        intList.Any(x => someInt == x); // Compliant (should raise S6617)
+        intList.Any(x => x.Equals(0)); // Compliant (should raise S6617)
+        intList.Any(x => 0.Equals(x)); // Compliant (should raise S6617)
+        intList.Any(x => x is 1); // Noncompliant FP (should raise S6617)
+        intList.Any(x => 1 is x); // Error [CS0150]
+        intList.Any(x => x is someInt); // Error [CS0150]
+        intList.Any(x => someInt is x); // Error [CS0150]
+
+        intList.Any(x => x == x); // Noncompliant
+        intList.Any(x => someInt == anotherInt); // Noncompliant
+        intList.Any(x => someInt == 0); // Noncompliant
+        intList.Any(x => 0 == 0); // Noncompliant
+
+        intList.Any(x => x.Equals(x)); // Noncompliant
+        intList.Any(x => someInt.Equals(anotherInt)); // Noncompliant
+        intList.Any(x => someInt.Equals(0)); // Noncompliant
+        intList.Any(x => 0.Equals(0)); // Noncompliant
+        intList.Any(x => x.Equals(x + 1)); // Noncompliant
+
+        intList.Any(x => x.GetType() == typeof(int)); // Noncompliant
+        intList.Any(x => x.GetType().Equals(typeof(int))); // Noncompliant FP
+        intList.Any(x => MyIntCheck(x)); // Noncompliant
```

Comment:
Add this type of check but without the argument for completion sake:

```csharp
intList.Any(MyIntCheck) // Noncompliant
```

same for string check